### PR TITLE
fix(#1623): Move querying metadata-validation to frontend service

### PR DESCRIPTION
**Pull request by MSzalowski**
## Summary
Haskell backend should be responsible only for fetching the DB-sync data. So I moved the querying the metadata-service logic to frontend that can consume 2 backends with ease and later on - react on socket events when appropriate designs on the cards are done.
That operation helps in decreasing the complexity level in Haskell backend making it easier to track and investigate errors around faulty SQL ([#ref](https://github.com/IntersectMBO/govtool/issues/1650)) or `Thread killed by timeout issues` - #1417

## List of changes

- Move metadata-validation service requests to frontend service

## Checklist

- [related issue](https://github.com/IntersectMBO/govtool/issues/1623)
- [x] My changes generate no new warnings
- [x] My code follows the [style guidelines](https://github.com/IntersectMBO/govtool/tree/main/docs/style-guides) of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the [changelog](https://github.com/IntersectMBO/govtool/blob/main/CHANGELOG.md)
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by MSzalowski** [comment]
After much time on investigation [this issue](https://github.com/IntersectMBO/govtool/issues/1650) I decided to not include fix for that in that PR and provide metadata from validation service once validated.

**Comment by MSzalowski** [comment]
This PR is the solution candidate as it introduces breaking change to the infrastructure.
I'll make more investigation around https://github.com/IntersectMBO/govtool/issues/1650 before merging this one.

**Review by placek — APPROVED** [review]
LGTM

File: govtool/frontend/src/services/requests/getDRepList.ts [hunk 1]
```diff
@@ -28,5 +34,38 @@ export const getDRepList = async ({
...
+  const validatedResponse = {
+    ...response.data,
+    elements: await Promise.all(
+      response.data.elements.map(async (drep) => {
+        if (drep.metadataStatus || drep.metadataValid) {
+          return {
+            ...drep,
+            metadataStatus: drep.metadataStatus,
+            metadataValid: drep.metadataValid,
+          };
```
Comment (jdyczka) [inline govtool/frontend/src/services/requests/getDRepList.ts:47]: Isn't this the same as just `return drep`?

File: govtool/frontend/src/services/requests/getProposals.ts [hunk 2]
```diff
@@ -29,5 +31,39 @@ export const getProposals = async ({
...
+  const validatedResponse = {
+    ...response.data,
+    elements: await Promise.all(
+      response.data.elements.map(async (proposal) => {
+        if (proposal.metadataStatus || proposal.metadataValid) {
+          return {
+            ...proposal,
+            metadataStatus: proposal.metadataStatus,
+            metadataValid: proposal.metadataValid,
+          };
```
Comment (jdyczka) [inline govtool/frontend/src/services/requests/getProposals.ts:44]: Same as before, `return proposal` should be fine

File: govtool/frontend/src/services/requests/getProposal.ts [hunk 3]
```diff
@@ -1,11 +1,34 @@
...
+    return data;
+  }
+  if (!data.url || !data.metadataHash) {
+    return data;
+  }
+
+  const validationResponse = await postValidate({
+    url: data.url,
+    hash: data.metadataHash,
+  });
```
Comment (jdyczka) [inline govtool/frontend/src/services/requests/getProposal.ts:22]: In other requests `standard` is specified. Isn't it necessary here?

**Review by jdyczka — APPROVED** [review]

**Pull request merged**